### ISSUE TITLE
script causes Safari to reach 100% CPU usage

### ISSUE THREAD

**Issue by fletom** [reek/anti-adblock-killer#1143]
After installing the anti-adblock-killer script using TamperMonkey for Safari, the CPU usage of the app reaches 100% while browsing around, and does not lower even when switching to another app. To test, I disabled the script (but left TamperMonkey itself enabled), restarted the browser, and the problem did not reoccur after hours of browsing. After re-enabling the script to double check, the strain on the CPU was back within a few minutes.

Mac OS 10.11.3 / Safari 9.0.3

**Comment by reek** [comment]
Hi,
Unfortunately I can't test Aak on Safari, because I use windows

**Comment by Giwayume** [comment]
You want your CPU to be at 100% utilization, that's a good thing. The problem comes in when the operating system does a poor job sharing that utilization evenly with all of your programs.

**Comment by fletom** [comment]
@Giwayume That's utter nonsense on both counts.

1) Nobody ever wants their CPU running at 100% utilization from a process stuck in a tight loop, draining their battery and spinning up their fans to the highest RPM. I don't know how you could possibly argue that's a good thing.

2) [Chrome is anywhere from 3x to 300x slower than Safari](https://blog.runspired.com/2016/03/25/the-chrome-distortion-chrome-alters-our-expectations-in-highly-negative-ways/).

**Comment by Giwayume** [comment]
1) Never did I say you want it to always be at 100% utilization. When it's actually performing the calculations you're requesting it to perform, it should get up to 100% utilization unless you're hampering it with a power plan. That's expected, and that's desired. Otherwise, why did you bother buying a CPU that powerful if you're not going to use it?

2) There are several different browser benchmarks, all with varying results in different categories, throwing someone's random blog article around isn't going to convince anyone. A blanket statement such as "Chrome is anywhere from 3x to 300x slower than Safari." is laughable.
http://web.basemark.com/
https://webkit.org/perf/sunspider/sunspider.html
http://peacekeeper.futuremark.com/

**Comment by fletom** [comment]
1) What else is "you want your CPU to be at 100% utilization" supposed to mean? We're talking about a browser here, not video transcoding. This issue, if you read it, is about constant 100% utilization, so why else would you chime in and say "that's a good thing"?

2) Peacekeeper? Deprecated. Sunspider? Safari is 5.25x faster according to [Digital Trends](http://www.digitaltrends.com/computing/best-browser-internet-explorer-vs-chrome-vs-firefox-vs-safari-vs-edge/). So that's one independently confirmed metric in which Chrome is "from 3x to 300x slower". I've never heard of Browsermark and don't have time to run it but I would be very surprised if Chrome wins. Safari is the faster browser. Strange how you can claim something completely false like "Chrome is a much superior browser performance-wise" with no data or references and then call it "laughable" when I actually link to a post from a respected JS dev to support the opposite.

Honestly if you don't have anything to contribute to this issue just don't comment. Your "wisdom" on CPU utilization and browser performance is irrelevant to this bug and would be just as irrelevant even if you weren't wrong.

**Comment by Giwayume** [comment]
You might want to calm down on the snarkiness. This place is not the youtube comments, and I'm one of the people here to help.

**Comment by Giwayume** [comment]
To fully recreate you scenario, we would need a full list of extensions you're using as well as other user scripts you're running. Also the model of your computer would help.

**Comment by fletom** [comment]
You're right, I shouldn't be snarky. I was mostly annoyed about being told which browser to use and found it condescending.

I am no longer using anti-adblock-killer because of this bug, but if I ever come back to it and experience the same issue I will definitely post more info about other extensions. I wasn't running any other TamperMonkey scripts.
